test/powershell/Modules/Microsoft.PowerShell.Management/Clear-Content.Tests.ps1

```diff
@@ -75,35 +77,46 @@ Describe "Clear-Content cmdlet tests" -Tags "CI" {
-      Get-Content -Path "TestDrive:/$file3" -Stream $streamName | Should -BeExactly $streamContent
-      Clear-Content -Path "TestDrive:/$file3" -Stream $streamName
-      Get-Content -Path "TestDrive:/$file3" | Should -BeExactly $content2
-      Get-Content -Path "TestDrive:/$file3" -Stream $streamName | Should -BeNullOrEmpty
-    }
-
-    It "the '-Stream' dynamic parameter is visible to get-command in the filesystem" -Skip:(!$IsWindows) {
-      try {
-        Push-Location -Path TestDrive:
-        (Get-Command Clear-Content -Stream foo).parameters.keys -eq "stream" | Should -Be "stream"
+    Context "Clear-Content should work with alternate data streams on Windows" {
+      It "Alternate streams should be cleared with Clear-Content on a file" -Skip:(!$IsWindows) {
+        # Make sure the test is set up correctly.
+        Set-Content           -Path "TestDrive:/$file3" -Stream $streamName -Value $streamContent
+        Get-Content           -Path "TestDrive:/$file3" -Stream $streamName | Should -BeExactly $streamContent
+        # Truncate the stream.
+        Clear-Content         -Path "TestDrive:/$file3" -Stream $streamName -ErrorAction Stop
+        # The stream should exist, but should have 0 length.
+        $result = Get-Item -Path "TestDrive:/$file3" -Stream $streamName
+        $result | Should -BeOfType System.Management.Automation.Internal.AlternateStreamData
+        $result.length | Should -Be 0
+      }
+      It "Alternate streams should be cleared with Clear-Content on a directory" -Skip:(!$IsWindows) {
+        # Make sure the test is set up correctly.
+        Set-Content           -Path "TestDrive:/$dirName" -Stream $streamName -Value $streamContent
+        # Truncate the stream.
+        Get-Content           -Path "TestDrive:/$dirName" -Stream $streamName | Should -BeExactly $streamContent
+        Clear-Content         -Path "TestDrive:/$dirName" -Stream $streamName -ErrorAction Stop
+        # The stream should exist, but should have 0 length.
+        $result = Get-Item -Path "TestDrive:/$dirName" -Stream $streamName
+        $result | Should -BeOfType System.Management.Automation.Internal.AlternateStreamData
+        $result.length | Should -Be 0
+      }
+      It "the '-Stream' dynamic parameter is visible to get-command in the filesystem" -Skip:(!$IsWindows) {
+        try {
+          Push-Location -Path TestDrive:
+          (Get-Command Clear-Content -Stream foo).parameters.keys -eq "Stream" | Should -BeExactly "Stream"
+        }
+        finally {
+          Pop-Location
+        }
       }
-      finally {
+      It "the '-Stream' dynamic parameter should not be visible to get-command in the function provider" {
+        Push-Location -Path function:
+        { Get-Command Clear-Content -Stream $streamName } |
+          Should -Throw -ErrorId "NamedParameterNotFound,Microsoft.PowerShell.Commands.GetCommandCommand"
         Pop-Location
```

Comment:
Please use try-finally.

Reply:
```
3ec98bbbe8 test/powershell/Modules/Microsoft.PowerShell.Management/Clear-Content.Tests.ps1 (Sergey Vasin        2018-04-10 06:47:39 +0300  99)     It "the '-Stream' dynamic parameter should not be visible to get-command in the function provider" {
2df9aac811 test/powershell/Modules/Microsoft.PowerShell.Management/Clear-Content.Tests.ps1 (Klaudia Algiz       2018-05-17 14:42:04 -0700 100)       Push-Location -Path function:
2df9aac811 test/powershell/Modules/Microsoft.PowerShell.Management/Clear-Content.Tests.ps1 (Klaudia Algiz       2018-05-17 14:42:04 -0700 101)       { Get-Command Clear-Content -Stream $streamName } |
2df9aac811 test/powershell/Modules/Microsoft.PowerShell.Management/Clear-Content.Tests.ps1 (Klaudia Algiz       2018-05-17 14:42:04 -0700 102)         Should -Throw -ErrorId "NamedParameterNotFound,Microsoft.PowerShell.Commands.GetCommandCommand"
2df9aac811 test/powershell/Modules/Microsoft.PowerShell.Management/Clear-Content.Tests.ps1 (Klaudia Algiz       2018-05-17 14:42:04 -0700 103)       Pop-Location
c6bd000e6d test/powershell/Modules/Microsoft.PowerShell.Management/Clear-Content.Tests.ps1 (James Truher [MSFT] 2017-03-06 13:57:09 -0800 104)     }
```

I'll fix this, because I reindented it to be part of a Context block to make it more clear that it needs to be run only on Windows. I'm tagging @kalgiz on this as well, since she's the original author of the code on the master branch (as the above `git blame` output says).

Klaudia: I'm sorry to bring up something from 2 1/2 years ago, and I don't know if this is something you've since learned or not. I'm sorry for bothering you if you have. If you haven't, though, I'd like to show you this:  here, if the test fails, the Pop-Location won't necessarily be called unless it's in a finally block.  If it's not called, it will mess up all subsequent tests in the test run (which assume they're located in the filesystem provider). With tests, it's best to look at the assumptions and make sure they're not violated. Otherwise, this is great!